### PR TITLE
Forbid passing a file base in child apps that are not alone

### PR DESCRIPTION
**Pull request by GiudGiud**
if you did that, it would write to the same file which was very cryptic when it crashed

### CODE REVIEW

**Comment by GiudGiud** [comment]
Failures unrelated

File: framework/src/outputs/FileOutput.C [hunk 1]
```diff
@@ -64,7 +64,14 @@ FileOutput::FileOutput(const InputParameters & parameters)
     _file_num = 0;
 
   if (isParamValid("file_base"))
+  { // Check that we are the only process or not a subapp
+    if (!_app.isUltimateMaster())
+      if (_app.multiAppNumber() > 0)
+        mooseError("A file base has been specified but there are multiple child apps for this "
+                   "MultiApp. The file base would be obeyed and all child apps would write to the "
+                   "same file. This is disallowed.");
```
Comment (joshuahansel) [inline framework/src/outputs/FileOutput.C:72]:
```suggestion
        mooseError("The parameter 'file_base' may not be specified for a child app when the MultiApp has multiple instances of the child app.");
```

Comment (joshuahansel) [reply framework/src/outputs/FileOutput.C:72]: Give the reason why as a comment - doesn't need to be in error message.

File: test/tests/outputs/exodus/tests [hunk 2]
```diff
@@ -111,6 +112,29 @@
     exodiff = 'exodus_elem_id_out.e'
 
     issues = '#23386'
-    requirement = "The system shall support restricting which extra element id data defined on the input mesh are outputted with the ExodusII format."
+    requirement = "The system shall support restricting which extra element id data defined on the "
+                  "input mesh are outputted with the ExodusII format."
+  []
+
+  [errors]
+    requirement = "The system shall error if "
```
Comment (joshuahansel) [inline test/tests/outputs/exodus/tests:120]:
```suggestion
    requirement = "The system shall report an error if "
```

File: framework/src/outputs/FileOutput.C (hunk 1 above)
Comment (joshuahansel) [reply framework/src/outputs/FileOutput.C:72]: Note this suggestion will require clang-formatting.

File: test/tests/outputs/exodus/tests [hunk 3]
```diff
@@ -111,6 +112,29 @@
...
+      expect_err = 'Output\(s\) do not exist \(must be variable, scalar, postprocessor, or vector postprocessor\):'
+
+      detail = "the supplied names for including or excluding variables is invalid."
+    []
+    [file_base_in_child_apps]
+      type = 'RunException'
+      input = 'exodus_elem_id.i'
+      cli_args = "MultiApps/ma/type=FullSolveMultiApp MultiApps/ma/input_files=exodus_elem_id.i "
+                 "MultiApps/ma/positions='0 0 0 1 1 0' MultiApps/ma/cli_args=Outputs/file_base=bad"
+      expect_err = 'A file base has been specified but there are multiple child apps for this MultiApp'
```
Comment (joshuahansel) [inline test/tests/outputs/exodus/tests:134]: Update this accordingly

File: test/tests/outputs/exodus/tests [hunk 4]
```diff
@@ -111,6 +112,29 @@
...
+
+      detail = "a file base is passed for file outputs and the child apps would be writing the same "
+               "file."
```
Comment (joshuahansel) [inline test/tests/outputs/exodus/tests:137]:
```suggestion
      detail = "a file base is supplied for a child app when there are multiple instances."
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: framework/src/outputs/FileOutput.C (hunk 1 above)
Comment (GiudGiud) [reply framework/src/outputs/FileOutput.C:72]: why would we not tell the user why we disallow it

Comment (joshuahansel) [reply framework/src/outputs/FileOutput.C:72]: IMO: it's not relevant to the user. All they need to know is: they can't do that. The technical reasons for why are for the developers. However, I don't feel strongly either way, so choose whichever you prefer.

Comment (joshuahansel) [reply framework/src/outputs/FileOutput.C:72]: In general, I just try to not give users more information than they need, to avoid confusion/intimidation.

Comment (GiudGiud) [reply framework/src/outputs/FileOutput.C:72]: I view it differently. If we re going to crash someone s simulation with an error message they need to know as much as possible. If they feel they have to dive into the code to understand we re only dev friendly not user friendly

Comment (GiudGiud) [reply framework/src/outputs/FileOutput.C:72]: now what I have now might not be optimal still

Comment (joshuahansel) [reply framework/src/outputs/FileOutput.C:72]: Updated suggestion

File: test/tests/outputs/exodus/tests [hunk 5]
```diff
@@ -111,6 +112,28 @@
...
+      expect_err = 'Output\(s\) do not exist \(must be variable, scalar, postprocessor, or vector postprocessor\):'
+
+      detail = "the supplied names for including or excluding variables is invalid."
+    []
+    [file_base_in_child_apps]
+      type = 'RunException'
+      input = 'exodus_elem_id.i'
+      cli_args = "MultiApps/ma/type=FullSolveMultiApp MultiApps/ma/input_files=exodus_elem_id.i "
+                 "MultiApps/ma/positions='0 0 0 1 1 0' MultiApps/ma/cli_args=Outputs/file_base=bad"
+      expect_err = 'A file base has been specified but there are multiple child apps for this MultiApp'
```
Comment (GiudGiud) [inline test/tests/outputs/exodus/tests:134]:
```suggestion
      expect_err = 'The parameter \'file_base\' may not be specified for a child app when the MultiApp has multiple instances of the child app'
```

File: framework/src/outputs/FileOutput.C [hunk 6]
```diff
@@ -64,7 +64,14 @@ FileOutput::FileOutput(const InputParameters & parameters)
     _file_num = 0;
 
   if (isParamValid("file_base"))
+  { // Check that we are the only process or not a subapp
```
Comment (joshuahansel) [inline framework/src/outputs/FileOutput.C:67]:
```suggestion
  {
    // Check that we are the only process or not a subapp
```

Comment (joshuahansel) [reply framework/src/outputs/FileOutput.C:67]: Sorry, just noticed this weird formatting

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: framework/src/outputs/FileOutput.C (hunk 6 above)
Comment (GiudGiud) [reply framework/src/outputs/FileOutput.C]: fixed

**Review by joshuahansel — APPROVED** [review]

**Comment by GiudGiud** [comment]
Thanks for the review!

**Pull request merged**